docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,90 @@
+## compilation-from-portable-pdb
+
+Compilation from portable PDBs today is not completely possible, but is desirable in order to help reconstruct a compilation from source provided via source link or embedded in a pdb. Motivation is derived from [roslyn 41395](https://github.com/dotnet/roslyn/issues/41395).
+
+This document is restricted to the following assumptions:
+
+1. The benefit is for builds with `-deterministic` and published to the symbol server.
+2. Source generator and analyzer references are not needed for this task. They may be useful, but are out of scope for this document.
```

Comment:
Will the analyzers and generators used be recorded? I 100% agree we don't need their actual references to form a functional compilation. Yet if we're recording information about compilation to create the source we should record that a given analyzer or generator was used. That can help for instance us route bugs to the appropriate generator for bad code or to an analyzer saying "hey you missed a case you were trying to prevent"

Reply:
We can include them in the set of metadata references if you think it's valuable. Note that we can't reconstruct a compilation that we could use to re-run the analyzers exactly as they were executed during build without also including additional files and perhaps other information the analyzers/generators need from build.

Reply:
Almost impossible, as we would have to find their transitive references as well, which is dangerous.

Reply:
I'm more concerned with just knowing which analyzer and SG run, not their transitive references. The assembly qualified name is essentially what I'm imagining here.

Reply:
I suggest we consider including analyzer/generator assembly refs separately from this change. The goal of this change is to enable reconstructing compilation specifically for the purpose of validation that a DLL was compiled from given sources and for running additional analyzers after build. The format of the PDB info can be expanded in future with more information to support more scenarios.